### PR TITLE
test(events): Add full coverage tests for TriggersHandler

### PR DESCRIPTION
**Pull request by nedimtokic**
Add tests for TriggersHandler to ensure full coverage. Tests now verify that dispatch_event/7 correctly builds the SimpleEvent struct and forwards it to Core.dispatch_event/3, including the edge case when policy_name is nil.
Also test that register_target/2 properly delegates to Core.

### CODE REVIEW

File: libs/astarte_events/test/triggers_handler/triggers_handler_test.exs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      simple_trigger_id: UUID.uuid4(:raw),
+      parent_trigger_id: UUID.uuid4(:raw)
+    }
+
+    {:ok, realm: realm, target: target}
+  end
+
+  describe "register_target/2" do
+    test "register_target/2 delegates to Core", %{realm: realm, target: target} do
+      Core |> expect(:register_target, fn _realm, ^target -> :ok end)
```
Comment (noaccOS) [inline libs/astarte_events/test/triggers_handler/triggers_handler_test.exs:48]: is this line needed? I believe it should work regardless
